js/views/modals/Settings/Store.js

```diff
@@ -126,9 +141,49 @@ export default class extends baseVw {
       'click .js-submitModByID': 'clickSubmitModByID',
       'click .js-save': 'save',
       'click .js-storeVerifiedOnly': 'onClickVerifiedOnly',
+      'click .js-applyToCurrent': 'clickApplyToCurrent',
+      'click .js-applyToCurrentConfirm': 'clickApplyToCurrentConfirm',
     };
   }
 
+  startProcessingTimer() {
+    if (!this.processingTimer) {
+      this.processingTimer = setTimeout(() => {
+        this.processingTimer = null;
+        this.setState({ isBulkCoinUpdating: false });
+      }, 500);
+    }
+  }
+
+  setState(state = {}, options = {}) {
+    // When the state is set to processing, start a timer so it's visible even if it's very short.
+    if (state.isBulkCoinUpdating) this.startProcessingTimer();
+
+    // If the state is set to stop processing, let the timer finish.
+    if (state.hasOwnProperty('isBulkCoinUpdating') &&
+      !state.isBulkCoinUpdating &&
+      this.processingTimer) {
+      delete state.isBulkCoinUpdating;
+    }
+
+    super.setState(state, options);
+  }
+
```

Comment:
@rmisio I'm not sure the above it the best way to handle preventing new state changes from stopping the spinner before the user has had a chance to see it, suggestions are welcome here.